BuildCMakeDocsTarget/action.yml

```diff
@@ -0,0 +1,53 @@
+name: Build CMake Documentation Target
+description: This action builds release documentation via a CMake target, zips the directory up, and modifies or creates a release to include the .zip file
+author: thebhef
+
+inputs:
+  doc-target:
+    description: the name of the CMake target
+    required: True
+  cmake-directory:
+    description: the directory CMake should use to build
+    required: True
+  doc-output:
+    description: the directory containing the output of the CMake build, relative to cmake-directory
+    required: True
+  build-version:
+    description: the version which should be used for the tag and the name of the zip file
+    required: True
```

Comment:
I'd maybe make these optional and provide defaults; I'm not sure we'll actually end up customizing them.